src/app/Models/Traits/SpatieTranslatable/HasTranslations.php

```diff
@@ -213,4 +204,51 @@ public function __call($method, $parameters)
                 break;
         }
     }
+
+    /**
+     * @param  \Illuminate\Database\Eloquent\Model  $model
+     * @param  array  $attributes
+     * @param  string  $locale
+     * @return \Illuminate\Database\Eloquent\Model
+     */
+    private static function fillModelWithTranslations($model, array $attributes, string $locale)
+    {
+        foreach ($attributes as $attribute => $value) {
+            // in case case it's an array, we will check if the keys of the array match the possible translation locales,
+            // if they do, we will set the attribute translations directly from the array.
+            if (is_array($value)) {
+                $possibleTranslations = array_keys($value);
+                $translatableLocales = array_keys($model->getAvailableLocales());
+
+                //if the array keys match the translatable locales (all keys must match some locale, ['en' => something, 'pt' => qualquer])
+                if ($possibleTranslations === array_intersect($possibleTranslations, $translatableLocales)) {
+                    $model->setTranslations($attribute, $value);
+                    continue;
+                }
+            }
+
+            $model->setTranslation($attribute, $locale, $value);
+        }
+
+        return $model;
+    }
+
+    /**
+     * @param  \Illuminate\Database\Eloquent\Model  $model
```

Comment:
All docblocks should have the description, that's the most important part.

```suggestion
    /**
     * Out of an array of attributes, only keep those that are NOT translatable.
     * 
     * @param  \Illuminate\Database\Eloquent\Model  $model
```